src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterUtilities.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Definition.BundleWrappers;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Parameters
+{
+    public class SearchParameterUtilities : ISearchParameterUtilities
+    {
+        private readonly ISearchParameterDefinitionManager _searchParameterDefinitionManager;
+        private readonly SearchParameterStatusManager _searchParameterStatusManager;
+
+        public SearchParameterUtilities(
+            SearchParameterStatusManager searchParameterStatusManager,
+            ISearchParameterDefinitionManager searchParameterDefinitionManager)
+        {
+            EnsureArg.IsNotNull(searchParameterStatusManager, nameof(searchParameterStatusManager));
+            EnsureArg.IsNotNull(searchParameterDefinitionManager, nameof(searchParameterDefinitionManager));
+
+            _searchParameterStatusManager = searchParameterStatusManager;
+            _searchParameterDefinitionManager = searchParameterDefinitionManager;
+        }
+
+        public async Task AddSearchParameterAsync(ITypedElement searchParam)
+        {
+            try
+            {
+                _searchParameterDefinitionManager.AddNewSearchParameters(new List<ITypedElement>() { searchParam });
+
+                var searchParameterWrapper = new SearchParameterWrapper(searchParam);
+                await _searchParameterStatusManager.AddSearchParameterStatusAsync(new List<string>() { searchParameterWrapper.Url });
```

Comment:
What does it mean if this line fails but the `AddNewSearchParameters` call succeeded? Will we be in an inconsistent state?

Reply:
If AddSearchParameterStatusAsync fails after the SearchParameter is added to the definition manager, then the parameter would not be usable at all.  No newly posted resources would use it for extraction, and no reindex job would pick it up either.   The client would need to PUT an update to the parameter to fix cause it to be reprocessed, or restart the FHIR service.   When the code is implemented to poll for new changes in SearchParameters, it could also fix issues such as this.  I will make a note in that user story.